{
    "EEG": [
        [
            "Task 3_P_20_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.get_power_spectrum(self.window_size)\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f7_alpha_mean = np.array(power_dict['F7_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 1_P_03_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.get_power_spectrum(self.window_size)\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f7_alpha_mean = np.array(power_dict['F7_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 4_P_16_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.removeBadCh()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 2_P_17_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.get_power_spectrum(self.window_size)\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f7_alpha_mean = np.array(power_dict['F7_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 5_P_17_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.get_power_spectrum(self.window_size)\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f7_alpha_mean = np.array(power_dict['F7_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 3_P_21_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.get_power_spectrum(self.window_size)\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f7_alpha_mean = np.array(power_dict['F7_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 2_P_16_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.removeBadCh()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 4_P_17_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.get_power_spectrum(self.window_size)\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f7_alpha_mean = np.array(power_dict['F7_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 1_P_10_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.removeBadCh()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 5_P_16_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.removeBadCh()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 5_P_05_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 320, in pre_process\n    self.filterData()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 73, in filterData\n    self.raw.crop(tmin=5, tmax=self.raw.times[-1] - 5)\n  File \"<decorator-gen-265>\", line 12, in crop\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 1512, in crop\n    raise ValueError(f\"tmin ({tmin}) must be less than tmax ({tmax})\")\nValueError: tmin (5) must be less than tmax (0.45999999999999996)\n"
        ],
        [
            "Task 1_P_01_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.get_power_spectrum(self.window_size)\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 183, in get_power_spectrum\n    f3_alpha_mean = np.array(power_dict['F3_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F3_alpha_mean'\n"
        ],
        [
            "Task 2_P_10_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.removeBadCh()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 104, in removeBadCh\n    ica = ICA(n_components=n_channels, random_state=97, verbose=False)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"<decorator-gen-296>\", line 10, in __init__\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 465, in __init__\n    raise ValueError(\nValueError: Selecting one component with n_components=1 is not supported\n"
        ],
        [
            "Baseline_P_14_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.get_power_spectrum(self.window_size)\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f7_alpha_mean = np.array(power_dict['F7_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 5_P_10_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.removeBadCh()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Baseline_P_11_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.get_power_spectrum(self.window_size)\n    ^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 2_P_19_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 183, in get_power_spectrum\n    f4_alpha_mean = np.array(power_dict['F4_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F3_alpha_mean'\n"
        ],
        [
            "Task 4_P_01_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f8_alpha_mean = np.array(power_dict['F8_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 2_P_18_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.get_power_spectrum(self.window_size)\n    ^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 3_P_10_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.get_power_spectrum(self.window_size)\n    ^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Baseline_P_09_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.get_power_spectrum(self.window_size)\n    ^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 2_P_11_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f8_alpha_mean = np.array(power_dict['F8_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 4_P_10_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.get_power_spectrum(self.window_size)\n    ^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 3_P_19_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 183, in get_power_spectrum\n    f4_alpha_mean = np.array(power_dict['F4_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F3_alpha_mean'\n"
        ],
        [
            "Task 5_P_08_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f8_alpha_mean = np.array(power_dict['F8_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 1_P_21_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 183, in get_power_spectrum\n    f4_alpha_mean = np.array(power_dict['F4_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F3_alpha_mean'\n"
        ],
        [
            "Baseline_P_10_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.get_power_spectrum(self.window_size)\n    ^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 3_P_02_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.get_power_spectrum(self.window_size)\n    ^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ],
        [
            "Task 4_P_02_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 183, in get_power_spectrum\n    f4_alpha_mean = np.array(power_dict['F4_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F3_alpha_mean'\n"
        ],
        [
            "Task 4_P_12_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 183, in get_power_spectrum\n    f4_alpha_mean = np.array(power_dict['F4_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F3_alpha_mean'\n"
        ],
        [
            "Task 1_P_07_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f8_alpha_mean = np.array(power_dict['F8_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Baseline_P_02_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f8_alpha_mean = np.array(power_dict['F8_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Baseline_P_08_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 204, in get_power_spectrum\n    f8_alpha_mean = np.array(power_dict['F8_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F7_alpha_mean'\n"
        ],
        [
            "Task 2_P_02_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 183, in get_power_spectrum\n    f4_alpha_mean = np.array(power_dict['F4_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F3_alpha_mean'\n"
        ],
        [
            "Task 5_P_02_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 322, in pre_process\n    self.saveDataframe()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 183, in get_power_spectrum\n    f4_alpha_mean = np.array(power_dict['F4_alpha_mean'])\n                             ~~~~~~~~~~^^^^^^^^^^^^^^^^^\nKeyError: 'F3_alpha_mean'\n"
        ],
        [
            "Task 1_P_20_EEG.csv",
            "Traceback (most recent call last):\n  File \"/Users/alberto/Desktop/SenseCobot/pre_process.py\", line 54, in preprocess_file\n    p.pre_process()\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 321, in pre_process\n    self.get_power_spectrum(self.window_size)\n    ^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/Analysis/EEG.py\", line 106, in removeBadCh\n    ica.apply(self.raw, verbose=False)\n  File \"<decorator-gen-303>\", line 10, in apply\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2237, in apply\n    out = meth(**kwargs)\n          ^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/preprocessing/ica.py\", line 2259, in _apply_raw\n    data = raw[picks, start:stop][0]\n           ~~~^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 846, in __getitem__\n    return self._getitem(item)\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 849, in _getitem\n    sel, start, stop = self._parse_get_set_params(item)\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/io/base.py\", line 779, in _parse_get_set_params\n    sel = _picks_to_idx(self.info, item[0])\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/alberto/Desktop/SenseCobot/venv/lib/python3.11/site-packages/mne/_fiff/pick.py\", line 1279, in _picks_to_idx\n    raise ValueError(\nValueError: No appropriate channels found for the given picks (array([], dtype=int64))\n"
        ]
    ]
}